src/functions/getNextUploadBatch/framework/handler.ts

```diff
@@ -32,19 +32,22 @@ export async function handler(event: APIGatewayEvent, fnCtx: Context): Promise<R
     return createResponse(validationResult.error, HttpStatus.BAD_REQUEST);
   }
 
-  await bootstrapConfig();
-  const batchPromise = getNextUploadBatch(batchSize, interfaceType);
-  await batchPromise.then((response) => {
-    // compress response
+  try {
+    await bootstrapConfig();
+    const response = await getNextUploadBatch(batchSize, interfaceType);
     nextBatchData = [
       ...response.map(((row) => {
         return gzipSync(JSON.stringify(row.test_result)).toString('base64');
       })),
     ];
-  }).catch((err) => {
-    console.error(err);
-  });
-  return createResponse(nextBatchData, HttpStatus.CREATED);
+    return createResponse(nextBatchData, HttpStatus.CREATED);
+  } catch (err) {
+    error('Unable to retrive a batch of results - ', enrichError(err, interfaceType, batchSize));
```

Comment:
spelling mistake - retrive instead of retrieve. I'm sure you just put them in there for me :)

Reply:
Since we expanded the `try{}catch` block saying that we were `Unable to retrieve a batch` may not always be the case(?) It could fail on getting config / stringify or gzipping etc IMHO
other than that it looks good

Reply:
I have updated it so it prints out err.message instead of a generic filler.